crates/runtime/src/dataconnector/debezium.rs

```diff
@@ -197,3 +196,104 @@ impl DataConnector for Debezium {
         Some(debezium_kafka.stream_changes())
     }
 }
+
+#[derive(Serialize, Deserialize)]
+struct DebeziumKafkaMetadata {
+    consumer_group_id: String,
+    topic: String,
+    primary_keys: Vec<String>,
+    schema_fields: Vec<change_event::Field>,
+}
+
+async fn get_metadata_from_accelerator(dataset: &Dataset) -> Option<DebeziumKafkaMetadata> {
+    let accelerated_metadata = AcceleratedMetadata::new(dataset).await?;
+    let metadata = accelerated_metadata.get_metadata().await?;
+    Some(metadata)
+}
+
+async fn set_metadata_to_accelerator(
+    dataset: &Dataset,
+    metadata: &DebeziumKafkaMetadata,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    let accelerated_metadata = AcceleratedMetadata::new_create_if_not_exists(dataset).await?;
+    accelerated_metadata.set_metadata(metadata).await
+}
+
+async fn infer_metadata_from_kafka(
```

Comment:
I don't think this is a great name. I wouldn't expect `infer_metadata_from_kafka` to read from the topic and then restart the kafka topic.

Reply:
I changed it to `get_metadata_from_kafka` - which has a nice parallel to `get_metadata_from_accelerator`